src/shipit_static_analysis/shipit_static_analysis/clang.py

```diff
@@ -43,6 +44,27 @@
     '--from-build', 'linux64-clang-tidy'
 ]
 
+__third_party = None
+
+
+def list_third_party(work_dir, path='tools/rewriting/ThirdPartyPaths.txt'):
+    '''
+    List third party directories using mozilla-central file
+    Uses a local cache to avoid reading every time the file
```

Comment:
Additionally, when we import new third party directories, we usually update `ThirdPartyPaths.txt` in the same patch, so we don't want to use a cached version otherwise we'll report all code issues from new third-party directories.